composer/trainer/trainer.py

```diff
@@ -1759,9 +1743,22 @@ def _eval_train_metrics(self):
                         # TODO: Detect if self.run_event(Event.AFTER_DATALOADER) changes the training
                         # data and if so print a warning that metrics may return unexpected results
                         with get_precision_context(self.state.precision):
-                            outputs, targets = self._original_model.validate(eval_microbatch)
-                            # Run in same precision context to avoid NaNs
-                            self.train_metrics.update(outputs, targets)
+                            if any(self._original_model.validate(eval_microbatch)):  # backwards compatibility check
```

Comment:
This would cause calling `validate` twice. How about removing `validate` from `CompsoerModel` entirely, and then doing a:
```
if hasattr(self._original_model, 'validate'):
    warnings.warn(DeprecationWarning(...)
    eval_outputs, target = self...
    for _, metric in..
        metric.update(eval_outputs, target)
```

Note: I think the metric updating step was missing in this code block (I added back in)